plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -0,0 +1,634 @@
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static java.lang.Math.abs;
+import static java.lang.String.join;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+import static org.testng.Assert.assertEquals;
+
+public class TestPinotIntegrationSmokeTest
+        // TODO extend BaseConnectorTest
+        extends AbstractTestQueryFramework
+{
+    private static final int MAX_ROWS_PER_SPLIT = 1000;
+    private static final String ALL_TYPES_TABLE = "alltypes";
+    private static final int DEFAULT_PINOT_LIMIT_FOR_BROKER_QUERIES = 10;
+    private static final int ALL_TYPES_TABLE_ROW_COUNT = 1000;
+    private static final int ALL_TYPES_TABLE_NULL_ROW_COUNT = 17;
+    private static final int ALL_TYPES_TABLE_ARRAY_NULL_ROW_COUNT = 13;
+    private static final String MIXED_CASE_TABLE = "mixed_case";
+    private static final List<List<Object>> MIXED_CASE_TABLE_RESULTS;
+    private static final String JSON_TABLE = "my_table";
+
+    static {
+        ImmutableList.Builder<List<Object>> mixedCaseResultSetBuilder = ImmutableList.builder();
+        for (long i = 0; i <= MAX_ROWS_PER_SPLIT; i++) {
+            mixedCaseResultSetBuilder.add(ImmutableList.builder()
+                    .add("string_" + i)
+                    .add(i)
+                    .add(Instant.now().truncatedTo(ChronoUnit.DAYS).plusMillis(i).toEpochMilli())
+                    .build());
+        }
+        MIXED_CASE_TABLE_RESULTS = mixedCaseResultSetBuilder.build();
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Random random = new Random(22);
+        TestingKafka kafka = closeAfterClass(TestingKafka.createWithSchemaRegistry());
+        kafka.start();
+        TestingPinotCluster pinot = closeAfterClass(new TestingPinotCluster(kafka.getNetwork()));
+        pinot.start();
+
+        // Create and populate the all_types topic and table
+        kafka.createTopic(ALL_TYPES_TABLE);
+
+        int step = 3;
+        int offset = 1;
+        checkState(ALL_TYPES_TABLE_ROW_COUNT > ALL_TYPES_TABLE_NULL_ROW_COUNT, "COUNT must be greater than NULL_COUNT");
+        checkState(ALL_TYPES_TABLE_NULL_ROW_COUNT > ALL_TYPES_TABLE_ARRAY_NULL_ROW_COUNT, "NULL_COUNT must be greater than ARRAY_NULL_COUNT");
```

Comment:
Add a code comment why

Reply:
Maybe instead of the check, initialize `ALL_TYPES_TABLE_ROW_COUNT = number of non null rows + number of null rows, etc`. That would remove need for these checks, a comment explaining them, and would make the for loops below more readable.

Reply:
Reduced to 11 rows (1 + default limit for broker queries) and have a single row for all null and array with mixed nulls.